apps/fyle/tasks.py

```
@@ -345,14 +379,323 @@ def import_and_export_expenses(report_id: str, org_id: str, is_state_change_even
+    """
+    Schedule expense group for later processing when task logs are no longer active
+    :param changed_expense_ids: List of expense IDs whose fund_source changed
+    :param workspace_id: Workspace id
+    :param report_id: Report id
+    :param affected_fund_source_expense_ids: Dict of affected fund sources and their expense ids
+    :return: None
+    """
+    logger.info("Scheduling for later processing for changed expense ids %s in workspace %s", changed_expense_ids, workspace_id)
+
+    # generate some random string to avoid duplicate tasks
+    hashed_name = hashlib.md5(str(changed_expense_ids).encode('utf-8')).hexdigest()[0:6]
```

Ruff (0.13.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)